diff --git a/papilio-prog/devlist.h b/papilio-prog/devlist.h
index 8e0856a..5d08bcb 100644
--- a/papilio-prog/devlist.h
+++ b/papilio-prog/devlist.h
@@ -9,4 +9,5 @@ const char fb_string[]={
 "04002093 6 XC6SLX16;"
 "04004093 6 XC6SLX25;"
 "04008093 6 XC6SLX45;"
+"0362E093 6 XC7A15T;"
 };
diff --git a/papilio-prog/progalgxc3s.cpp b/papilio-prog/progalgxc3s.cpp
index 9ee2738..399521a 100644
--- a/papilio-prog/progalgxc3s.cpp
+++ b/papilio-prog/progalgxc3s.cpp
@@ -49,6 +49,7 @@ ProgAlgXC3S::ProgAlgXC3S(Jtag &j, IOBase &i, int fam)
     case 0x13: /* XC3SAN*/
     case 0x1c: /* SC3SADSP*/
     case 0x20: /* XC6S*/
+    case 0x36: /* XC7*/
       tck_len = 16;
       array_transfer_len = 16;
       break;
